hls4ml/converters/pyg_to_hls.py

```diff
@@ -0,0 +1,215 @@
+from __future__ import print_function
+import os
+
+import torch
+from hls4ml.converters.pytorch_to_hls import PyTorchModelReader
+from hls4ml.utils.config import create_vivado_config
+from hls4ml.model.hls_layers import HLSType, IntegerPrecisionType, FixedPrecisionType
+from hls4ml.model.hls_model import HLSModel_GNN
+
+
+class PygModelReader(PyTorchModelReader):
+    def __init__(self, config):
+        super().__init__(config)
+        self.n_node = config['n_node']
+        self.n_edge = config['n_edge']
+        self.node_dim = config['node_dim']
+        self.edge_dim = config['edge_dim']
+
+    def get_weights_data(self, layer_name, var_name, module_name=None):
+        data = None
+
+        # Parameter mapping from pytorch to keras
+        torch_paramap = {
+            # Conv
+            'kernel': 'weight',
+            # Batchnorm
+            'gamma': 'weight',
+            'beta': 'bias',
+            'moving_mean': 'running_mean',
+            'moving_variance': 'running_var'}
+
+        if var_name not in list(torch_paramap.keys()) + ['weight', 'bias']:
+            raise Exception('Pytorch parameter not yet supported!')
+
+        if module_name is not None:
+            if var_name in list(torch_paramap.keys()):
+                var_name = torch_paramap[var_name]
+
+            try:
+                data = self.state_dict[module_name + '.' + layer_name + '.' + var_name].numpy().transpose()
+            except KeyError:
+                data = self.state_dict[module_name + '.layers.' + layer_name + '.' + var_name].numpy().transpose()
+
+        else:
+            if var_name in list(torch_paramap.keys()):
+                var_name = torch_paramap[var_name]
+
+            data = self.state_dict[layer_name + '.' + var_name].numpy().transpose()  # Look at transpose when systhesis produce lousy results. Might need to remove it.
+
+        return data
+
+def pyg_to_hls(model, forward_dict, graph_dims,
+               activate_final = None,
+               fixed_precision_bits=16,
+               fixed_precision_int_bits=6,
+               int_precision_bits=16,
+               int_precision_signed=False,
+               output_dir = None,
+               reuse = 1):
```

Comment:
We should pass in a single `HLSConfig` object instead of each of these explicit arguments. See how it's done with `keras_to_hls`:
https://github.com/fastmachinelearning/hls4ml/blob/master/hls4ml/converters/keras_to_hls.py#L193
or `pytorch_to_hls`:
https://github.com/Duchstf/hls4ml/blob/onnx_dev_pytorch/hls4ml/converters/pytorch_to_hls.py#L106